### ISSUE TITLE
[High] Unwanted solutions due to overflow

### ISSUE THREAD

**Issue by ShreyanJabade** [fvutils/pyvsc#81]
Hello,
On using any constants (eg. integers) inside a constraint, unwanted solutions are also returned due to overflow. In the snippet below, for constraint a x b == 50, solutions like a = 182 and b = 155 are also getting returned because 182 x 155 has the same LSBs as 50.

I request you to look into this aspect.
Thank you

```
import vsc
import random
random.seed(0)

@vsc.randobj
class Test:
    def __init__(self):
        self.a = vsc.rand_uint8_t()
        self.b = vsc.rand_uint8_t()

    @vsc.constraint
    def test_c(self):
        self.a*self.b == 50

inst = Test()
inst.randomize()
print("a =", inst.a, "b =", inst.b)
print("Binary representation of a*b:", f"{inst.a*inst.b:b}")
print("Binary representation of 10:", f"{50:b}")
```

Output is:

```
a = 182 b = 155
Binary representation of a*b: 110111000110010
Binary representation of 10: 110010
```

**Comment by mballance** [comment]
After investigating, I found that both this issue and #82 are due to context-dependent expression widths not being properly accounted for. In the example above, the width of the multiplication result is 8 bits (the max width of the two operands). However, the context is 32-bit due to the presence of the literal. The expected result is that the width of the multiplication result will be 32-bit as well.
I've made a fix in 0.4.3 that resolves this for the testcases in #81 and #82. I'll leave this open until you can confirm that it works for you.

**Comment by ShreyanJabade** [comment]
There are no more unwanted solutions due to overflow. Thank you for your consideration. I checked the distribution in this case and it is similar to #83. Have commented it there as well. Apologies for the belated reply, somehow there were no notifications for issue replies. 

```
import vsc
import matplotlib.pyplot as plt

@vsc.randobj
class Test:
    def __init__(self):
        self.a = vsc.rand_uint8_t()
        self.b = vsc.rand_uint8_t()

    @vsc.constraint
    def test_c(self):
        self.a*self.b == 60

inst = Test()

a_vals = []
for i in range(1000):
    inst.randomize()
    a_vals.append(inst.b)

plt.hist(a_vals)
plt.show()
```

Output is:
![image](https://user-images.githubusercontent.com/58798198/123587864-881cf500-d804-11eb-98c6-2d84935edfb6.png)
